haskellPackages: update stackage and hackage

maralorn opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #327219. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

alexfmpe committed:
haskellPackages.math-functions: remove override

alexfmpe committed:
haskellPackages.vector-hashtables: remove override

maralorn committed:
haskellPackages: stackage LTS 22.29 -> LTS 22.31

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

maralorn committed:
all-cabal-hashes: 2024-07-14T21:17:20Z -> 2024-07-31T18:11:52Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

maralorn committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

AndersonTorres commented:
Please, can you verify how to remove the reference to cedille?

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

ymeister committed:
haskellPackages.patch: fix build on js backend

ymeister committed:
haskellPackages.gi-gtk: fix build on ghc > 9.6

github-actions[bot] committed:
Merge master into haskell-updates

ymeister committed:
haskellPackages.os-string: mark os-string as GHC-9.10 Core library

github-actions[bot] committed:
Merge master into haskell-updates

aforemny committed:
haskellPackages.dunai: unbreak

dunai by default depends on transformers <6.0.0. The list-transformer
flag has to be enabled to build against newer versions of transformers.

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

alexfmpe committed:
haskellPackages: disable tests on aarch64-darwin that require postgresql

github-actions[bot] committed:
Merge master into haskell-updates

raehik committed:
haskellPackages.bytezap: unbreak

raehik committed:
haskellPackages: edit raehik package maintainership

rvl committed:
haskellPackages.tasty: Drop overrides for 1.5.1

tasty-1.5.1 is now properly deprecated on Hackage.

https://github.com/UnkindPartition/tasty/issues/426#issuecomment-2257860103

rvl committed:
haskellPackages.monad-schedule: Fix eval

rvl committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

rvl committed:
haskellPackages.doctest: Skip failing cabal-doctest tests

The (experimental) cabal-doctest executable was added in 0.22.5, and
isn't yet installed by default.

rvl committed:
haskellPackages.feedback: Fix build

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge pull request #333379 from rvl/haskell-updates-doctest

haskellPackages.doctest: Skip failing cabal-doctest tests

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskellPackages.xnobar: Disable on darwin

maralorn committed:
Merge pull request #332093 from ymeister/haskell-updates@gi-gtk

haskellPackages.gi-gtk: fix build on ghc > 9.6

maralorn committed:
Merge pull request #333386 from rvl/haskell-updates-feedback

haskellPackages.feedback: Fix build

maralorn committed:
Merge branch 'raehik/unbreak-bytezap' into haskell-updates

maralorn committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

maralorn committed:
Merge pull request #333316 from alexfmpe/haskell-postgresql-aarch-darwin

haskellPackages: disable tests on aarch64-darwin that require postgresql

maralorn committed:
Merge pull request #332090 from ymeister/haskell-updates@patch

haskellPackages.patch: fix build on js backend

maralorn committed:
Merge pull request #332123 from ymeister/haskell-updates@os-string

haskellPackages.os-string: mark os-string as GHC-9.10 Core library

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
git-annex: update sha256 for 10.20240731

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
stack: get latest version building

maralorn commented:
@Anton-Latukha @sorki we have a build error with hnix-store on the branch. Can you maybe look into that or advise what the cause and the easiest fix might be?
https://hydra.nixos.org/build/269662115/nixlog/1/tail

eldritch-cookie committed:
haskellPackages.avro: drop unnecessary patch

alexfmpe committed:
haskellPackages.stan: remove override

alexfmpe committed:
haskellPackages.posix-api: jailbreak to unbreak

alexfmpe committed:
haskellPackages.hakyll-filestore: remove override

alexfmpe committed:
haskellPackages.streamly-bytestring: remove override

alexfmpe committed:
haskellPackages.blucontrol: remove override

alexfmpe committed:
haskellPackages.bench: remove override

alexfmpe committed:
haskellPackages.comfort-blas: remove override

alexfmpe committed:
haskellPackages.storablevector: remove override

alexfmpe committed:
haskellPackages.shh-extras: remove override

alexfmpe committed:
haskellPackages.haskell-ci: remove jailbreak

alexfmpe committed:
haskellPackages.diagrams-gtk: remove override

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
Merge pull request #334977 from alexfmpe/posix-api

haskellPackages.posix-api: jailbreak to unbreak

maralorn committed:
haskellPackages: fix formatting

maralorn committed:
Merge pull request #334976 from alexfmpe/old-overrides

haskellPackages: remove several old overrides

maralorn committed:
Merge pull request #335094 from eldritch-cookie/drop-avro-patch

haskellPackages.avro: drop unnecessary patch

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.hnix-store-remote: restrict to 0.6.*

hnix 0.17.0 requires hnix-store-remote < 0.7 at the moment, so we'll
stay with this version until a new hnix version is released.

I noticed that there is something sort of wrong with hnix in nixpkgs. We
have a constraint from Stackage on hnix-store-core (and no other
package) which doesn't match the rest of the hnix packages as we need to
use hnix-store-core == 0.6.*. It may be prudent to check why this
constraint is in Stackage and whether we can downgrade to 0.6.*. It
seems to me that the store 0.8.* packages are not ready yet, i.e. no
support from hnix. hnix-store-core == 0.7.* seems to be a strange
version that hasn't really any other matching packages.

github-actions[bot] committed:
Merge master into haskell-updates

rvl committed:
haskellPackages.quickspec: Unmark broken

Package version 2.1.5 failed to build and was therefore marked broken.

Package version 2.2 has fixed the twee-lib dependency version
constraint and can build successfully.

rvl committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

rvl committed:
haskellPackages.quickcheck-state-machine: Fix build failure in tests

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
Merge pull request #335548 from rvl/haskell-updates-quickspec

haskellPackages.quickspec: Unmark broken

sorki commented:
> @Anton-Latukha @sorki we have a build error with hnix-store on the branch. Can you maybe look into that or advise what the cause and the easiest fix might be? https://hydra.nixos.org/build/269662115/nixlog/1/tail

Looks like @sternenseemann already fixed the issue (I just noticed after removing some restrictions on `core`). I might still adjust it so the `-core` and `-remote` point to latest versions and the additional older ones are used for `hnix`. I have a WIP PR for `hnix` but it will take some time to finalize and get a release out (and it seems couple more changes to `hnix-store` would be nice along with a minor release).

The reason for `hnix-store-core 0.7` release was a fix for `cachix`. Same with `0.8` where NAR serialization is split into `hnix-store-nar` so it can receive updates independently.

maralorn committed:
haskell.packages.ghc92.haskell-language-server: Fix build by disabling plugins

maralorn committed:
haskellPackages: Fix eval errors

maralorn committed:
Merge branch 'master' into haskell-updates

maralorn committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1808369](https://hydra.nixos.org/eval/1808369) of nixpkgs commit [f3583fa](https://github.com/NixOS/nixpkgs/commits/f3583fa38bdb980eeae71916bb5dd877bbfe6a55) as of 2024-08-19 22:22 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates
